### ISSUE TITLE
Alternative APK download

### ISSUE THREAD

**Issue by IzzySoft** [vauvenal5/yaga#19]
Would you mind making the app available outside the walled garden of Google Play – so that e.g. owners of Huawei devices and others without access to that place get a chance to use it? You could e.g. attach the `*.apk` files to their corresponding tags/releases, or bring it to [F-Droid](https://f-droid.org/). Thanks in advance for considering!

**Comment by vauvenal5** [comment]
Hi!
I am already in the process of being on-boarded onto F-Droid, [here](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/7200) the link to the merge request.

There are two open issues I have to solve, a minor problem with my handling of version numbers in the build, and the bigger problem that I am using GIT LFS for all png involved in the project. Regarding the GIT LFS issue I am still waiting for a comment from side of F-Droid.

I will also look into attaching the apk files to the releases.

**Comment by IzzySoft** [comment]
> I am already in the process of being on-boarded onto F-Droid

Ah, apologies – I must have missed that (with 300+ open issues in RFP, one looses track sometimes). And I don't look into MRs at fdroiddata regularly, being neither Android dev nor packager :wink:

> Regarding the GIT LFS issue I am still waiting for a comment from side of F-Droid.

Well, in that case it might be a good idea to have your app [in my repo](https://apt.izzysoft.de/fdroid) until the open points are solved? I'd just need the APKs for that as I cannot build myself.

> I will also look into attaching the apk files to the releases.

Great! Looking forward to that then. Just leave a comment here please when they're available, so I get a notice and can pick up. Thanks a lot!

**Comment by vauvenal5** [comment]
Github release is up.

**Comment by IzzySoft** [comment]
Thanks! Ah, now I see what the issue is with LFS. I've cloned your repo to copy Fastlane structures over, but all binary files (here: images) are broken. Well… Good that my updater will fetch that via Github's API, that still seems to work :smile:

I'd send you a PR with a minor update for Fastlane, but I'm not sure if that would break as well. Hence let me just include the "diff" here:

* `de/short_description.txt`: "eine Nextcloud-first Galerie-App"
* for the full description: I don't know if you use Fastlane Supply for Google Play (just found out Google is crazy there, saying "de" is not a valid language, insisting on `de-DE`, `de-AT`, `de-CH` etc) – and if you do, how Google treats "simple inline HTML tags"; but with F-Droid, such are possible. So for listing in my repo (your app will [show up here](https://apt.izzysoft.de/fdroid/index/apk/com.github.vauvenal5.yaga) with the next sync in about 22h) I've slightly adjusted the description, using `<a href` for the links and `<i>` for the app name. The latter cannot hurt, as in the worst case they just strip it. So I just mention it and leave the decision to you :wink:

PS: What's not quite clear from the description, does the app *only* deal with images stored in a Nextcloud – or with the local ones as well? Also, is there a minimum Nextcloud version required?

**Comment by vauvenal5** [comment]
There were a few changes since yesterday. I untracked files needed for fastlane and for the build from git lfs to re-enabled the F-Droid build. However there seems to be a general problem currently since the `apt-get update` command in the F-Droid pipeline is failing.

Regarding `de` fastlane files, I will look into it the coming days. Yes, I am using fastlane to supply Google Play as well as Github Releases.

Edit: The app is also able to display images from your internal storage, however, not internally formatted SD cards are not yet supported. Regarding minimum NC version, honestly I don't know. Currently it uses only webdav APIs so depends on how long those have been stable.

**Comment by IzzySoft** [comment]
> I untracked files needed for fastlane and for the build from git lfs

Thanks! Though I'm not sure if those are the only ones stopping the F-Droid build.

> the `apt-get update` command in the F-Droid pipeline is failing.

I've noticed that for some days now, too. Need to check if someone's on it already, as that's completely out of my field. I've asked on IRC.

> Yes, I am using fastlane to supply Google Play

Looks like Google has some stupid rules established concerning this. From another project, just dealing with those:

    /usr/local/share/gems/gems/google-api-client-0.38.0/lib/google/apis/core/http_command.rb:227:in `check_status': badRequest: The requested language is not currently supported: de. (Google::Apis::ClientError)

In short: Google refuses ISO-639 languages ("xx" instead of "xx-XX") as "unsupported". A possible work-around would be using `de-DE` (which then might end up de-AT users not being served this), or having `de-DE` symlinking to `de` and ignoring those Fastlane errors.

> The app is also able to display images from your internal storage…

Maybe that should be mentioned in the app description? As for WebDAV: there should be no issues then, agreed – as long as you don't call to routines provided by the NC gallery app, which might change with version updates (I've seen the latter for several apps, eg. bookmarks).

**Comment by vauvenal5** [comment]
- [ ] `de` fastlane files
- [ ] improve description: mention local storage

> NC gallery app

I do intend to add albums support etc once this has been resolved: https://github.com/nextcloud/photos/issues/205

Until then plenty of other tasks and features still open ;)

**Comment by IzzySoft** [comment]
Should you run into issues with `fastlane supply` on the `de` directory, I've just learned F-Droid itself would resolve `de-DE` to `de` implicitly in their client. Just let me know then that I establish a symlink so that works with my "web client" as well, which would *not* assume `de` if the user explicitly asks for `de-DE` only – and neither would use `de-DE` if not explicitly requested (e.g. by browser preference); thus a user with "de-AT,de" would receive en-US (the fallback) instead if only de-DE were present…

Google is an annoyance in more than one way…
